### PR TITLE
#165673703 Slug implementation

### PR DESCRIPTION
**Pull request by ZackAtama**
#### What does this PR do?
- Improves slug implementation
- Replaces id with slug for getting, updating and deleting an article

#### Description of the task to be completed?
- The current slug implementation adds more calls to the database and the id property is not needed as per API outputs

#### What are the relevant pivotal tracker stories
- [#165673703](https://www.pivotaltracker.com/story/show/165673703)

Reviewers
@kalsmic @bisonlou @nadralia  @llwasampijja @JamesMudidi

### CODE REVIEW

File: authors/apps/articles/models.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 from authors.apps.authentication.models import User
 
 class Article(models.Model):
-    slug = models.CharField(max_length=120, unique=True)
+    slug = models.CharField(max_length=100, unique=True)
     title = models.CharField(max_length=100)
```
Comment (bisonlou) [inline authors/apps/articles/models.py:6]: Slug and article title cannot have the same length. A slug should be longer than the article title.

File: authors/apps/articles/serializers.py [hunk 2]
```diff
@@ -5,7 +5,6 @@ class ArticleSerializer(serializers.ModelSerializer):
     """This serializer requests and creates a new article"""
     class Meta:
         fields = [
-            'id',
             'slug',
```
Comment (nadralia) [inline authors/apps/articles/serializers.py:8]: @AtamaZack NiceWork i see you have the read_only_field empty you could add the slug and author . or just remove it if there is no field that needs to be added there.

**Review by nadralia — CHANGES_REQUESTED** [review]

**Review by bisonlou — CHANGES_REQUESTED** [review]

File: authors/apps/articles/serializers.py (hunk 2 above)
Comment (ZackAtama) [reply authors/apps/articles/serializers.py:8]: Thanks, on it.

File: authors/apps/articles/models.py (hunk 1 above)
Comment (ZackAtama) [reply authors/apps/articles/models.py:6]: Thank you, on it.

**Review by JamesMudidi — APPROVED** [review]
Nice work zack, after implementing the changes requested by @nadralia and @bisonlou, we are good to go

File: authors/apps/articles/models.py (hunk 1 above)
Comment (bisonlou) [reply authors/apps/articles/models.py:6]: Well done @AtamaZack

**Review by bisonlou — APPROVED** [review]

**Review by nadralia — APPROVED** [review]

File: authors/apps/articles/extra_methods.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import random
+import string
+from django.utils.text import slugify
+from .models import Article
+
+def random_string_generator(size=10, chars=string.ascii_lowercase + string.digits):
+    return ''.join(random.choice(chars) for _ in range(size))
+
+def create_slug(object_model, title):
```
Comment (kalsmic) [inline authors/apps/articles/extra_methods.py:9]: Well done @AtamaZack, It's good you have created the "create_slug" method, now we can also use it in our tests to create a slug for an article

**Review by kalsmic — APPROVED** [review]

**Review by llwasampijja — APPROVED** [review]

**Pull request merged**